marshmallow/schema.py

```diff
@@ -513,7 +509,7 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
                     **kwargs
                 )
             except ValidationError as error:
-                errors = self._marshal.errors
+                errors = marshal.errors
```

Comment:
The marshaller passes its errors to the `messages` argument of `ValidationError()`, so we should be able to access it via `error.messages` instead of reaching back through the marshaller.